pepagent/pepagent.py

```diff
@@ -392,15 +610,142 @@ def _get_namespace_proj_anno(self, namespace: str = None) -> dict:
+                ANNO_COL: result[4],
             }
 
         return res_dict
 
+    def project_exists(
+        self,
+        namespace: str = DEFAULT_NAMESPACE,
+        name: str = None,
+        tag: str = DEFAULT_TAG,
+    ) -> bool:
+        """
+        Checking if project exists in the database
+        :param namespace: project namespace
+        :param name: project name
+        :param tag: project tag
+        :return: Returning True if project exist
+        """
+        if name is None:
+            _LOGGER.error(f"Name is not specified")
+            return False
+
+        sql = f"""SELECT {ID_COL} from {DB_TABLE_NAME} 
+                    WHERE {NAMESPACE_COL} = %s AND
+                          {NAME_COL} = %s AND 
+                          {TAG_COL} = %s;"""
+
+        if self.run_sql_fetchone(sql, namespace, name, tag):
+            return True
+        else:
+            return False
+
+    def project_exists_by_registry(
+        self,
+        registry_path: str,
+    ) -> bool:
+        """
+        Checking if project exists in the database
+        :param registry_path: project registry path
+        :return: Returning True if project exist
+        """
+
+        reg = ubiquerg.parse_registry_path(
+            registry_path,
+            defaults=[
+                ("namespace", DEFAULT_NAMESPACE),
+                ("item", None),
+                ("tag", DEFAULT_TAG),
+            ],
+        )
+        namespace = reg["namespace"]
+        name = reg["item"]
+        tag = reg["tag"]
+
+        if self.project_exists(namespace=namespace, name=name, tag=tag):
+            return True
+        else:
+            return False
+
+    def project_status(
+        self,
+        namespace: str = DEFAULT_NAMESPACE,
+        name: str = None,
+        tag: str = DEFAULT_TAG,
+    ) -> str:
+        """
+        Retrieve project status by providing name, namespace and tag
+        :param namespace: project registry - will return dict of project annotations
+        :param name: project name in database. [required if registry_path does not specify]
+        :param tag: tag of the projects
+        :return: status
+        """
+        sql_q = f"""
+                select ({ANNO_COL}->>'status') as status
+                        from {DB_TABLE_NAME}
+                            WHERE {NAMESPACE_COL}=%s AND
+                                {NAME_COL}=%s AND {TAG_COL}=%s;
+                """
+
+        if not name:
+            _LOGGER.error(
+                "You haven't provided neither registry_path or name! Execution is unsuccessful. "
+                "Files haven't been downloaded, returning empty dict"
+            )
+            return "None"
+
+        if not self.project_exists(namespace=namespace, name=name, tag=tag):
+            _LOGGER.error("Project does not exist, returning None")
+            return "None"
+
+        result = self.run_sql_fetchone(sql_q, namespace, name, tag)
+
+        return result[0]
```

Comment:
Example of where `run_sql_fetchone` should just return the first result it finds. Instead of extracting the first index of `result`.

Reply:
So.. run_sql_fetchone returns set of first result, e.g. when you want to get name, namespace, and tag, then you will get result ('name_', 'namespace_','tag_'). In this case we need just item 0.

Reply:
okkk, I see where is the problem

Reply:
I see. I think I was getting confused as to why `self.run_sql_fetchone` was returning lists instead of `tuple`s

Reply:
I think id doesn't make big difference

Reply:
return empty dict, if nothing was found is more convenient . I think so..

Reply:
There was an issue with list, that can cause an error. I fixed it

Reply:
More on `tuple`s v `list`s: https://stackoverflow.com/a/1708538

At the end of the day, it seems like `tuple`s are better if the data doesn't need to be mutated - they are faster. But that's about it.